src/conditions/check_fields.rs

```diff
@@ -326,6 +329,57 @@ impl CheckFieldsPredicate for ExistsPredicate {
 
 //------------------------------------------------------------------------------
 
+#[derive(Debug, Clone)]
+struct IpCidrPredicate {
+    target: Atom,
+    cidrs: Vec<IpCidr>,
+}
+
+impl IpCidrPredicate {
+    pub fn new(
+        target: String,
+        arg: &CheckFieldsPredicateArg,
+    ) -> Result<Box<dyn CheckFieldsPredicate>, String> {
+        let cidr_strings = match arg {
+            CheckFieldsPredicateArg::String(s) => vec![s.clone()],
+            CheckFieldsPredicateArg::VecString(ss) => ss.clone(),
+            _ => {
+                return Err(
+                    "ip_cidr_contains predicate requires a string or list of string argument"
+                        .to_owned(),
+                )
+            }
+        };
+        let cidrs = match cidr_strings.iter().map(|s| IpCidr::from_str(s)).collect() {
+            Ok(v) => v,
+            Err(e) => return Err(format!("Invalid IP CIDR: {}", e)),
+        };
+        let target = target.into();
+        Ok(Box::new(Self { target, cidrs }))
+    }
+}
+
+impl CheckFieldsPredicate for IpCidrPredicate {
+    fn check(&self, event: &Event) -> bool {
+        match event {
+            Event::Log(l) => l.get(&self.target).map_or(false, |v| {
+                let v = v.to_string_lossy();
+                let ip_addr = match IpAddr::from_str(&v) {
+                    Ok(addr) => addr,
+                    Err(e) => {
+                        error!("Invalid IP \"{}\": {}", v, e);
```

Comment:
I don't think an error is appropriate here. The other predicate that does conversion (`EqualsPredicate`) simply returns `false` with no notice when conversion fails.

Reply:
Good call - removing that error